prog/ai/inference_endpoint_nexus.rb

```diff
@@ -0,0 +1,132 @@
+# frozen_string_literal: true
+
+require "forwardable"
+
+require_relative "../../lib/util"
+
+class Prog::Ai::InferenceEndpointNexus < Prog::Base
+  subject_is :inference_endpoint
+
+  extend Forwardable
+  def_delegators :inference_endpoint, :replicas, :load_balancer, :private_subnet, :project
+
+  semaphore :destroy
+
+  def self.assemble_with_model(project_id:, location:, name:, model_id:,
+    replica_count: 1, is_public: false)
+    model = Option::AI_MODELS.detect { _1["id"] == model_id }
+
+    fail "Model with id #{model_id} not found" unless model
+
+    assemble(
+      project_id: project_id,
+      location: location,
+      name: name,
+      boot_image: model["boot_image"],
+      vm_size: model["vm_size"],
+      storage_volumes: model["storage_volumes"],
+      model_name: model["model_name"],
+      engine: model["engine"],
+      engine_params: model["engine_params"],
+      replica_count: replica_count,
+      is_public: is_public
+    )
+  end
+
+  def self.assemble(project_id:, location:, boot_image:, name:, vm_size:, storage_volumes:, model_name:,
+    engine:, engine_params:, replica_count:, is_public:)
+    unless (project = Project[project_id])
+      fail "No existing project"
+    end
+
+    Validation.validate_location(location)
+    Validation.validate_name(name)
+    Validation.validate_vm_size(vm_size)
+    fail "Invalid replica count" unless replica_count.is_a?(Integer) && replica_count > 0 && replica_count < 10
```

Comment:
It becomes bit difficult to read unless statements with multiple conditions. How about;

```
fail if !replica_count.is_a?(Integer) || replica_count <= 0 || replica_count >= 10
```
or

```
fail if !replica_count.is_a?(Integer) || replica_count < 1 || replica_count > 9
```